Reduction of Raw Efields (#71)
ECP-WarpX/artemis: Docs/source/usage/parameters.rst

        * ``<reduced_diags_name>.reduction_type`` (`string`)
            The type of reduction to be performed. It must be either ``Maximum``, ``Minimum`` or
            ``Integral``.
            ``Integral`` computes the spatial surface or volume integral, depending on the choice

>             of the integration_type, of the function defined in the parser by summing its value on

Reviewer: ```suggestion
            of the ``integration_type``, of the function defined in the parser by summing its value on
```

Reviewer: The new documentation should ideally come under a new Type called 
`RawEFieldReduction` unlike, where it is now, FieldReduction.

The main difference here FieldReduction first interpolates fields to the cell-center, where RawEFieldReduction integrates the fields from the staggered yee-grid

            all grid points and multiplying the result by the area or volume of a cell.
            Please be also aware that measuring maximum quantities might be very noisy in PIC
            simulations.

        The only output column is the reduced value.

        Note that the fields are averaged on the cell centers before the reduction is performed.

>         * ``<reduced_diags_name>.integration_type`` (`string`)
>            The type of integration to be performed. It must be either ``surface`` or ``volume``.

Reviewer: ```suggestion
        * ``<reduced_diags_name>.integration_type`` (`string`)
           The type of integration to be performed. It must be either ``surface`` or ``volume``. Note that for the surface integral, the user-defined parser for the plane should be less than one cell size in thickness in the direction of the surface normal. Also, the direction of the ``surface_normal`` must be specified for surface integration.
```

Author: In this case we should not write about the directionality of surface normal, it is just X or Y or Z, I think. (Please see my next comment) The first sentence I agree with.

        * ``<reduced_diags_name>.surface_normal`` (`string`)

>            The surface on which the surface integration is required. It must be either ``X``, ``Y`` or ``Z``.

Reviewer: what about negatve direction surface normal?

Suppose we have a box, then we would have both positive and negative surface normals.

Author: Hi Reva, I haven't accounted for the direction here. 
I thought the field vector (e.g. E=i Ex+j Ey + k Ez) would account for the direction. 

If Ez is negative everywhere on the plane, then its value integrated over the plane will be negative. E.g. if on half of the surface of the Y (or XZ) plane, Ez is 1 and other half it is -1, surface integral will be zero. The normals are not really normals in the sense of directionality, but just a mean to indicate whether it is a X, Y, or a Z plane.

Reviewer: Discussed offline about directionality for surface normal.

This is not for normal fields only, it sums the values of raw E-fields on a given plane.
If the user needs to compute surface integral normal to the plane, then they can do that in post-processing. So this will be clarified in the documentation

Author: Clarified this in the documentation. Please feel free to make changes.

        Note that the fields are averaged on the cell centers before the reduction is performed.

    * ``RawEFieldReduction``

>         This type is ONLY for the E-field at the respected staggering and executes the parser 
>         as a function of (x,y,z) to reduce Ex, Ey, and Ez.

Reviewer: ```suggestion
        This type is ONLY for the E-field at their respective staggering on the Yee-grid (or the type of grid used in the
        simulation) and executes the ``reduced_function`` which is a user-defined analytic function as given below.

        * ``<reduced_diags_name>.reduced_function(x,y,z)`` (`string`)
              An analytic function used to select the region over which the electric fields will be reduced using
              the ``reduction_type`` described below.               
        
```

            of the ``integration_type``, of the function defined in the parser by summing its value on
            all grid points and multiplying the result by the area or volume of a cell.
            Please be also aware that measuring maximum quantities might be very noisy in PIC
            simulations.

        The output columns correspond to the timestep counter, physical time, and reduced values of Ex, Ey, Ez components.

        * ``<reduced_diags_name>.integration_type`` (`string`)
           The type of integration to be performed. It must be either ``surface`` or ``volume``.
           Note that the surface integral provides the integrated normal as well as the
           traverse fields over the surface. For the surface integral, the user-defined parser
           for the plane should be less than one cell size in thickness in the direction of the surface normal.
           Also, if the specified surface has an edge that overlaps with the domain boundary but the edge parallel
           to it does not overlap with the domain boundary, then exclude the edge that overlaps with the domain boundary
           while defining the surface.

>            See examples in Examples/Tests/Macroscopic_Maxwell/inputs_3d_surfaceIntegralDiagnostic_tests_cubic and
>            Examples/Tests/Macroscopic_Maxwell/inputs_3d_surfaceIntegralDiagnostic_tests_noncubic for ways to
>            define a full, half, and a quarter plane.

Reviewer: ```suggestion
```

if you want to given an example, typically, we write the example in the documentation of the specific part. 
Also, if for some reason, we decide to remove those tests, we may not remember to change the documentation accordingly :-)
